### PR TITLE
8276932: G1: Annotate methods with override explicitly in g1CollectedHeap.hpp

### PR DESCRIPTION
**Pull request by rvansa**
Hi all,

This pull request contains a backport of commit[ 02f79008](https://github.com/openjdk/jdk/commit/02f79008828b3dcce3bd6492efaa43e99376c0c5) from the [openjdk/jdk](https://git.openjdk.org/jdk) repository.

The commit being backported was authored by Albert Mingkun Yang on 15 Nov 2021 and was reviewed by Thomas Schatzl and Stefan Johansson.

The backport is not clean; some method return types differ in later JDK versions.

Thanks!

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [ ] [JDK-8276932](https://bugs.openjdk.org/browse/JDK-8276932) needs maintainer approval

### Issue
 * [JDK-8276932](https://bugs.openjdk.org/browse/JDK-8276932): G1: Annotate methods with override explicitly in g1CollectedHeap.hpp (**Enhancement** - P4)

### Reviewers
 * [Paul Hohensee](https://openjdk.org/census#phh) (@phohensee - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk17u-dev.git pull/2830/head:pull/2830` \
`$ git checkout pull/2830`

Update a local copy of the PR: \
`$ git checkout pull/2830` \
`$ git pull https://git.openjdk.org/jdk17u-dev.git pull/2830/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 2830`

View PR using the GUI difftool: \
`$ git pr show -t 2830`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk17u-dev/pull/2830.diff">https://git.openjdk.org/jdk17u-dev/pull/2830.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk17u-dev/pull/2830#issuecomment-2315656916)

### CODE REVIEW

**Review by phohensee — APPROVED** [review]

**Comment by GoeLin** [comment]
There are GHA errors.

**Comment by rvansa** [comment]
@GoeLin Thanks, one failure was failure to fetch JTReg (infra problem), I am re-running the failed ClhsdbSource test since I find it difficult to relate to changes in this PR.

**Comment by rvansa** [comment]
The checks are now green (no changes, just rerun).

**Comment by phohensee** [comment]
You need to ask for maintainer approval using "/approval request <text>".

**Comment by rvansa** [comment]
@phohensee Actually I've added the `jdk17u-fix-request` label in JIRA, but it was removed by @GoeLin - so is this practically a rejection?

**Pull request closed without merging**
